controllers/articles-controllers.js

```diff
@@ -1,14 +1,26 @@
-const { checkArticleIdExists } = require("../utils.js");
+const { checkArticleIdExists,
+        checkTopicExists } = require("../utils.js");
 const { fetchArticleById, 
         fetchArticles,
         fetchArticleComments,
         createArticleComment, 
         updateArticle } = require("../models/articles-models.js")
 
 module.exports.getArticles = (req, res, next) => {
-    return fetchArticles().then((articles) => {
-        res.status(200).send({articles});
-    }).catch(next);
+    if (Object.keys(req.query).length !== 0){
+        const { topic } = req.query;
+        const checkTopicExistsQuery = checkTopicExists(topic);
+        const articlesQuery = fetchArticles(topic);
+        const queryArray = [articlesQuery, checkTopicExistsQuery];
+        return Promise.all(queryArray)
+        .then((articles) => {
```

Comment:
I think you should rename this `articles` variable as it describes an array where only the first value are the articles 🔍

Reply:
I've renamed this variable to make it more clear that it is the results of the Promise.all and that the articles list is the first result of that array. Refactored.